pkg/contract/allocations.go

```diff
@@ -235,3 +235,39 @@ func GetEVMSubnetGenesisNativeMinterAdmin(
 	}
 	return getGenesisNativeMinterAdmin(app, network, genesisData)
 }
+
+func ContractAddressIsInBlockchainGenesis(
+	app *application.Avalanche,
+	network models.Network,
+	chainSpec ChainSpec,
+	contractAddress common.Address,
+) (bool, error) {
+	genesisData, err := GetBlockchainGenesis(
+		app,
+		network,
+		chainSpec,
+	)
+	if err != nil {
+		return false, err
+	}
+	if !utils.ByteSliceIsSubnetEvmGenesis(genesisData) {
```

Comment:
nit: can we fix inverse logic and return ContractAddressIsInGenesisData for the happy path and `else` return `false, fmt.Errorf`

Reply:
this will change the way this is done in other parts of the package. this is a check for a precondition
on the input data, and sometimes there are more checks before or after the processing itself, and
returning here is more related to the condition check.